app/src/main/java/com/islaharper/dawnofandroid/presentation/screens/welcome/WelcomeScreen.kt

```diff
@@ -1,12 +1,176 @@
 package com.islaharper.dawnofandroid.presentation.screens.welcome
 
-import androidx.compose.material.Text
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
 import androidx.navigation.NavHostController
+import com.google.accompanist.pager.ExperimentalPagerApi
+import com.google.accompanist.pager.HorizontalPager
+import com.google.accompanist.pager.HorizontalPagerIndicator
+import com.google.accompanist.pager.PagerState
+import com.google.accompanist.pager.rememberPagerState
+import com.islaharper.dawnofandroid.R
+import com.islaharper.dawnofandroid.domain.model.OnBoardingPage
+import com.islaharper.dawnofandroid.navigation.Screen
+import com.islaharper.dawnofandroid.ui.theme.AndroidGreen
+import com.islaharper.dawnofandroid.ui.theme.Neutral90
+import com.islaharper.dawnofandroid.ui.theme.PADDING_SMALL
+import com.islaharper.dawnofandroid.ui.theme.PADDING_XLARGE
+import com.islaharper.dawnofandroid.util.Constants.ONBOARDING_LAST_PAGE
+import com.islaharper.dawnofandroid.util.Constants.ONBOARDING_PAGE_COUNT
 
+@OptIn(ExperimentalPagerApi::class)
 @Composable
 fun WelcomeScreen(
     navHostController: NavHostController,
+    welcomeViewModel: WelcomeViewModel = hiltViewModel(),
 ) {
-    Text(text = "Welcome Placeholder")
+    val pages = listOf(
```

Comment:
Does having this typed as list (instead of `ImmutableList`) and not remembered cause unnecessary recomposition?

Reply:
Change made in latest commit.